src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs

```diff
@@ -878,5 +878,93 @@ private void MinerCreateNewBlockValidity()
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockCheckNodeConsensusTipIsCorrect()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode miner = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+
+                miner.WithWallet();
+                builder.StartAll();
+                
+                TestHelper.MineBlocks(miner, 5);
+
+                Assert.Equal(5, miner.FullNode.ConsensusManager().Tip.Height);
+            }
+        }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockCheckPeerHasNewBlock()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode miner = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+                
+                CoreNode[] syncers = new CoreNode[]
+                {
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD(),
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD(),
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD()
+                };
+
+                miner.WithWallet();
+                builder.StartAll();
+                
+                foreach (CoreNode syncer in syncers)
+                {
+                    miner.CreateRPCClient().AddNode(syncer.Endpoint, true);
+                }
+
+                Assert.True(syncers.All(x => x.FullNode.ConsensusManager().Tip.Height == 0));
+
+                TestHelper.MineBlocks(miner, 3);
+
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[0]));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[1]));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[2]));
+
+                Assert.True(syncers.All(x => x.FullNode.ConsensusManager().Tip.Height == 3));
+            }
+        }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockNotPushedToConsensusCode_SupercededByBetterBlockOnReorg_InitialBlockRejected()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode node1 = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+                CoreNode node2 = builder.CreateStratisPowNode(KnownNetworks.RegTest).NotInIBD();
+
+                node1.WithWallet();
+                node2.WithWallet();
+
+                builder.StartAll();
+                
+                node1.CreateRPCClient().AddNode(node2.Endpoint, true);
+                TestHelper.MineBlocks(node1, 5);
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(node1, node2));
+                node1.CreateRPCClient().RemoveNode(node2.Endpoint);
+
+                // Node1 new block not pushed to consensus code.
+                Transaction tx = node1.FullNode.Network.CreateTransaction();
```

Comment:
No need to create a new tx here, if the list is empty the method will just ignore it ;)